testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergRedirectionToHive.java

```diff
@@ -352,21 +368,18 @@ public void testInformationSchemaColumns()
         createHiveTable(hiveTableName, false);
 
         // via redirection with table filter
-        // TODO: support redirects from Iceberg to Hive
-        assertQueryFailure(() -> onTrino().executeQuery(
+        assertThat(onTrino().executeQuery(
                 format("SELECT * FROM iceberg.information_schema.columns WHERE table_schema = '%s' AND table_name='%s'", schemaName, tableName)))
-                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): Not an Iceberg table: " + schemaName + "." + tableName);
+                .containsOnly(
+                        row("iceberg", schemaName, tableName, "nationkey", 1, null, "YES", "bigint"),
+                        row("iceberg", schemaName, tableName, "name", 2, null, "YES", "varchar(25)"),
+                        row("iceberg", schemaName, tableName, "comment", 3, null, "YES", "varchar(152)"),
+                        row("iceberg", schemaName, tableName, "regionkey", 4, null, "YES", "bigint"));
 
         // test via redirection with just schema filter
         assertThat(onTrino().executeQuery(
                 format("SELECT * FROM iceberg.information_schema.columns WHERE table_schema = '%s'", schemaName)))
-                .containsOnly(
-                        // TODO report table's columns via redirect to Hive
-//                        row("iceberg", schemaName, tableName, "nationkey", 1, null, "YES", "bigint"),
-//                        row("iceberg", schemaName, tableName, "name", 2, null, "YES", "varchar(25)"),
-//                        row("iceberg", schemaName, tableName, "comment", 3, null, "YES", "varchar(152)"),
-//                        row("iceberg", schemaName, tableName, "regionkey", 4, null, "YES", "bigint")
-                        /**/);
+                .hasNoRows();
```

Comment:
With the changes introduced in https://github.com/trinodb/trino/commit/a2ea77b6f4f0f208b00ca4c6d2e99eb6814538b8 `io.trino.plugin.iceberg.TrinoHiveCatalog#listTables(io.trino.spi.connector.ConnectorSession, java.util.Optional<java.lang.String>)` is listing *only* Iceberg hive tables.

This is breaking the expected functionality for the queries which make use just of the schema filter.

I tend to duplicate the content of `io.trino.plugin.iceberg.TrinoCatalog#listTables(io.trino.spi.connector.ConnectorSession, io.trino.spi.connector.SchemaTablePrefix)` method for this specific purpose and retrieve all the metastore tables instead of retrieving, but I am also rather reluctant to expose two methods similar methods in the TrinoCatalog (which may be misleading for the users of its API).

@losipiuk @findepi @phd3